samples/nrf9160/azure_fota/src/main.c

```diff
@@ -244,8 +248,25 @@ void main(void)
 	modem_configure();
 	k_sem_take(&network_connected_sem, K_FOREVER);
 
-	static struct azure_iot_hub_config config = {
+#if IS_ENABLED(CONFIG_DEVICE_ID_USE_HW_ID)
+	char device_id[HW_ID_LEN];
+
+	err = hw_id_get(device_id, ARRAY_SIZE(device_id));
+	if (err) {
+		LOG_ERR("Failed to retrieve device ID");
+		return;
+	}
+#endif
+
+	struct azure_iot_hub_config config = {
 		.use_dps = IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS),
+#if IS_ENABLED(CONFIG_DEVICE_ID_USE_HW_ID)
+		.device_id = {
+			.ptr = device_id,
```

Comment:
```suggestion
			.ptr = IS_ENABLED(CONFIG_DEVICE_ID_USE_HW_ID) ? device_id : NULL,
```

I think I would prefer something like this instead of chopping up the code with the preprocessor statement.
If the compiler doesn't complain when `CONFIG_DEVICE_ID_USE_HW_ID` is disabled that is :smile:

Reply:
Agree it would look better, but doesn't work as `device_id` is undeclared in this case.